mac/mcompile/keymap.h

```diff
@@ -0,0 +1,549 @@
+#pragma once
+#ifndef KEYMAP_H
+#define KEYMAP_H
+
+#include <iostream>
+#include <Carbon/Carbon.h>
+#include <string>
+#include <vector>
+
+#include "../../common/include/km_u16.h"
+
+enum ShiftState {
+  Base         = 0,                   // 0
+  Shft         = 1,                   // 1
+  Ctrl         = 2,                   // 2
+  ShftCtrl     = Shft | Ctrl,         // 3
+  Menu         = 4,                   // 4 -- NOT USED
+  ShftMenu     = Shft | Menu,         // 5 -- NOT USED
+  MenuCtrl     = Menu | Ctrl,         // 6
+  ShftMenuCtrl = Shft | Menu | Ctrl,  // 7
+  Xxxx         = 8,                   // 8
+  ShftXxxx     = Shft | Xxxx,         // 9
+};
+
+#define VK_SPACE    0x20
+#define VK_COLON	  0xBA
+#define VK_EQUAL	  0xBB
+#define VK_COMMA	  0xBC
+#define VK_HYPHEN   0xBD
+#define VK_PERIOD	  0xBE
+#define	VK_SLASH	  0xBF
+#define VK_ACCENT	  0xC0
+#define VK_LBRKT	  0xDB
+#define VK_BKSLASH	0xDC
+#define VK_RBRKT	  0xDD
+#define VK_QUOTE	  0xDE
+#define VK_xDF		  0xDF
+#define VK_OEM_102  0xE2  //  "<>" or "\|" on RT 102-key kbd.
+
+#define VK_DIVIDE   0x6F
+#define VK_CANCEL   3
+#define VK_DECIMAL  0x2E
+
+#define VK_OEM_CLEAR      0xFE
+#define VK_LSHIFT         0xA0
+#define VK_RSHIFT         0xA1
+#define VK_LCONTROL       0xA2
+#define VK_RCONTROL       0xA3
+#define VK_LMENU          0xA4
+#define VK_RMENU          0xA5
+
+#define VK_SHIFT          0x10
+#define VK_CONTROL        0x11
+#define VK_MENU           0x12
+#define VK_PAUSE          0x13
+#define VK_CAPITAL        0x14
+// Map of all US English virtual key codes that we can translate
+const KMX_DWORD KMX_VKMap[] = {
```

Comment:
This should be defined in the .cpp and only declared as an extern here:

```suggestion
extern const KMX_DWORD KMX_VKMap[];
```